pkgs/development/compilers/gcc/11/default.nix

```
@@ -96,8 +103,13 @@ stdenv.mkDerivation ({
 
   inherit patches;
 
-  outputs = [ "out" "man" "info" ] ++ lib.optional (!langJit) "lib";
-  setOutputFlags = false;
+  outputs = [ "out" "man" "info" ]
+            ++ lib.optional (!langJit) "lib"
+            ++ lib.optionals enableLibGccOutput [ "libgcc" ]
+            ++ lib.optionals enableExternalBootstrap [ "checksum" ]
+            # the plugins reference both $out and $lib, so we cannot put them in either
+            ++ lib.optionals (enableExternalBootstrap && enablePlugin) [ "plugin" ];
```

```suggestion
  ++ lib.optional (!langJit) "lib"
  ++ lib.optionals enableLibGccOutput [ "libgcc" ]
  ++ lib.optionals enableExternalBootstrap [ "checksum" ]
  # the plugins reference both $out and $lib, so we cannot put them in either
  ++ lib.optionals (enableExternalBootstrap && enablePlugin) [ "plugin" ];
```

I don't think we want that many custom named outputs.

---

> I don't think we want that many custom named outputs.

it's ok as long as there is a good reason and there is

---

We could merge the `checksum` output into `out`.  But the `libgcc` and `plugin` have to be separate in order to prevent circular dependencies (which Nix does not allow).

---

I think the `plugin` output can be merged into `$out`.  I'm testing that right now.

---

> I think the `plugin` output can be merged into `$out`. I'm testing that right now.

Yep, it works.  I've pushed a commit which removes the `plugin` output.

So currently we have: `out` `lib` `libgcc` and `checksum`.

It's possible to eliminate `checksum`, but the other three have to be kept separate.

---

I'm going to resolve this since I was able to eliminate the `plugin` output, and three of the four remaining outputs (`out`, `lib`, and `libgcc`) have to remain separate to prevent circular dependencies.

The fourth output `checksum` should never be used by anything other than `test/stdenv/default.nix##gcc-stageCompare`, so I think a custom output name is okay there.